core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaPageSource.java

```diff
@@ -272,11 +273,11 @@ private void addTablesRecords(QualifiedTablePrefix prefix)
     {
         Set<SchemaTableName> tables = listTables(session, metadata, accessControl, prefix);
         Set<SchemaTableName> views = listViews(session, metadata, accessControl, prefix);
-        // TODO (https://github.com/trinodb/trino/issues/8207) define a type for materialized views
+        Set<SchemaTableName> materializedViews = listMaterializedViews(session, metadata, accessControl, prefix);
 
-        for (SchemaTableName name : union(tables, views)) {
+        for (SchemaTableName name : union(union(tables, views), materializedViews)) {
             // if table and view names overlap, the view wins
-            String type = views.contains(name) ? "VIEW" : "BASE TABLE";
+            String type = materializedViews.contains(name) ? "MATERIALIZED VIEW" : views.contains(name) ? "VIEW" : "BASE TABLE";
```

Comment:
This line is a little hard to understand quickly. It would be better to use `if` in my opinion.